From c62a0cf96a8059c2cef5acea94bbe07855aafaad Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@kernel.org>
Date: Thu, 16 Feb 2023 12:10:05 -0500
Subject: [PATCH 30/35] dm: add cond_resched() to dm_wq_requeue_work()

Otherwise the while() loop in dm_wq_requeue_work() can result in a
"dead loop" on systems that have preemption disabled. This is
particularly problematic on single cpu systems.

Fixes: 8b211aaccb915 ("dm: add two stage requeue mechanism")
Cc: stable@vger.kernel.org
Signed-off-by: Mike Snitzer <snitzer@kernel.org>
---
 drivers/md/dm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/md/dm.c b/drivers/md/dm.c
index b424a6ee2..3aa4952ef 100644
--- a/drivers/md/dm.c
+++ b/drivers/md/dm.c
@@ -1008,6 +1008,7 @@ static void dm_wq_requeue_work(struct work_struct *work)
 		io->next = NULL;
 		__dm_io_complete(io, false);
 		io = next;
+		cond_resched();
 	}
 }
 
-- 
2.39.2.501.gd9d677b2d8

